src/test/scala/simd/PETest.scala

```diff
@@ -0,0 +1,206 @@
+package simd
+
+import chisel3._
+import org.scalatest.flatspec.AnyFlatSpec
+import chiseltest._
+import scala.math.BigInt
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.Tag
+
+// import lib for random test data generation
+import scala.util.Random
+import scala.math.pow
+
+// Post-Processing unit golden model trait
+trait HasPostProcessingGoldenModel {
+  def postProcessingGoldenModel(
+      input: Int,
+      inputZp: Byte,
+      outputZp: Byte,
+      multiplier: Int,
+      shift: Byte, // values between 0-63
+      maxInt: Byte,
+      minInt: Byte,
+      doubleRound: Boolean
+  ): Byte = {
+
+    // input zero-point adjustment
+    var adjustedInput = input - inputZp
+
+    // multiplication
+    val var0: Long = adjustedInput.toLong * multiplier.toLong
+
+    // shift & round
+    var var1: Int = (var0 >> (shift - 1)).toInt
+
+    if (doubleRound) {
+      if (var1 >= 0)
+        var1 += 1
+      else
+        var1 -= 1
+    }
+    var1 = var1 >> 1
+
+    // output zero-point adjustment
+    var1 = var1 + outputZp
+
+    // clamping
+    if (var1 > maxInt)
+      var1 = maxInt
+    if (var1 < minInt)
+      var1 = minInt
+
+    val result: Byte = var1.toByte
+    result
+  }
+}
+
+// Random test data generation trait
+trait HasRandomTestGen {
+  def RandomTestGen(): (Int, Byte, Byte, Int, Byte, Byte, Byte, Boolean) = {
+    // Initialize random seed
+    val random = new Random(System.currentTimeMillis())
+
+    // Generate random values
+    val input: Int = random.nextInt(math.pow(2, 28).toInt)
```

Comment:
Why $2^{28}$ only? Shouldn't the random be $2^{32}-1$? Or if it's signed then it's like from negative to positive $2^{31}-1$ right?

Reply:
need to hit the range of (-128,127)